optional-container-engine/bookshelf-frontend.yaml

```diff
@@ -0,0 +1,79 @@
+# Copyright 2015 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License
+
+# This file configures the bookshelf application frontend. The frontend serves
+# public web traffic.
+
+# The bookshelf frontend replication controller ensures that at least 3
+# instances of the bookshelf app are running on the cluster.
+# For more info about Pods see:
+#   https://cloud.google.com/container-engine/docs/pods/
+# For more info about Replication Controllers:
+#   https://cloud.google.com/container-engine/docs/replicationcontrollers/
+apiVersion: v1
+kind: ReplicationController
+metadata:
+  name: bookshelf-frontend
+  labels:
+    app: bookshelf
+    tier: frontend
+spec:
+  replicas: 3
+  selector:
+    app: bookshelf
+    tier: frontend
+  template:
+    metadata:
+      labels:
+        app: bookshelf
+        tier: frontend
+    spec:
+      containers:
+      - name: bookshelf-app
+        image: gcr.io/gaev2-samples/bookshelf
+        imagePullPolicy: Always
```

Comment:
I guess it could be removed, but it's something that tripped me up when I forgot a config thing. If you feel like users won't need it I can remove it.